### ISSUE TITLE
Re-ordering Patterns

### ISSUE THREAD

**Issue by olympianoiseco** [olympianoiseco/Patterning-2-Public#48]
• Ability to re-order patterns for FF / RW buttons and for Pattern Select views in RECORD/SONG

**Comment by nathankopp** [comment]
AGREED! This seems like it would be pretty easy, and would make managing longer songs MUCH easier!

For reordering, I'd recommend maybe allwoing drag-and-drop to the top part of the "song" tab. Maybe have it work similar to the (x) button, where you hold the move button and then drag a pattern circle to a new location in the list.

For rename, maybe on the same song tab the user can double-tap one of the circles to bring up a modal box for entering the new name.
